### ISSUE TITLE
Add support for websockets

### ISSUE THREAD

**Issue by tomsej** [roapi/roapi#40]
It would be great to add support for websockets too - https://actix.rs/docs/websockets/. Thanks a lot!

**Comment by novemberkilo** [comment]
I'd like to tackle this if that's appropriate. My plan is:

- turn on the `ws` feature flag for `axum` https://docs.rs/axum/latest/axum/#feature-flags
- write an integration test, possibly using `curl`

**Comment by novemberkilo** [comment]
@houqp my initial thoughts about this issue appear to have been naive. Simply enabling the feature in axum doesn't get us very far. Won't we actually need to support a websocket endpoint? Essentially something like `roapi-ws` that is a websockets version of `roapi-http`? This seems like a much bigger piece of work.

**Comment by houqp** [comment]
yes, a new websocket endpoint will be needed, but I think we don't need an extra `roapi-ws` binary and it can be folded into roapi-http.

**Comment by novemberkilo** [comment]
That makes sense thanks @houqp 

I might start by sketching out the experience of using the current API from a websockets client. If you have any thoughts on this please let me know?

**Comment by houqp** [comment]
One way to do that is to map the current REST endpoints into commands sent as websocket messages. We will only need a single websocket endpoint to handle all the commands.
